src/main/scala/exercises/04_Options.scala

```diff
@@ -34,16 +34,31 @@ object Options {
    * - does not exist: 					"not existing"
    */
   def roomState(rooms: Map[Int, Option[String]], room: Int): String = {
-    error("Fix me")
+      rooms.get(room) match {
+        case None => "not existing"
+        case Some(state) => state match{
+          case None => "empty"
+          case Some("locked") => "not available"
+          case Some(n) => n
+        }
+      }
+ 
   }
 
+
   /**
    * Calculate the total amount of people in all rooms
 	 *
    * Hint: define a helper function that computes a room's occupancy
    * to convert a possible numeric String (e.g. Some("12")) to an integer
    */
   def totalPeopleInRooms(rooms: Map[Int, Option[String]]): Int = {
-    error("Fix me")
+    var count = 0;
+    for(a <- 1 until 6){
+      val state = roomState(rooms, a);
+      if(!(state == "not existing" || state == "empty" || state == "not available")) 
+        count = count + state.toInt;
+    }
```

Comment:
This can also be done using the fold function. I suspect it is more scala-y to use fold it is more like the functional style to use higher order functions over loops.